evm/src/trace/decoder.rs

```diff
@@ -39,6 +41,78 @@ impl CallTraceDecoder {
     /// as DSTest-style logs.
     pub fn new() -> Self {
         Self {
+            // TODO: These are the Ethereum precompiles. We should add a way to support precompiles
+            // for other networks, too.
+            precompiles: [
+                precompile(
+                    1,
+                    "ecrecover",
+                    [
+                        ParamType::FixedBytes(32),
+                        ParamType::Uint(256),
+                        ParamType::Uint(256),
+                        ParamType::Uint(256),
+                    ],
+                    [ParamType::Address],
+                ),
+                precompile(2, "keccak", [ParamType::Bytes], [ParamType::FixedBytes(32)]),
+                precompile(3, "ripemd", [ParamType::Bytes], [ParamType::FixedBytes(32)]),
+                precompile(4, "identity", [ParamType::Bytes], [ParamType::Bytes]),
+                precompile(
+                    5,
+                    "modexp",
+                    [
+                        ParamType::Uint(256),
+                        ParamType::Uint(256),
+                        ParamType::Uint(256),
+                        ParamType::Bytes,
+                    ],
+                    [ParamType::Bytes],
+                ),
```

Comment:
I am not sure this will decode because the format is a little strange, so it will show up as `PRECOMPILE::modexp(big ass blob)`

Reply:
I wrote a HexDisplay helper on the node PR
https://github.com/gakonst/foundry/blob/243c5048016e068b8d09667899910591a9f73d00/node/src/eth/util.rs
that only prints certain lengths, maybe useful here as well?